src/components/example/certificate/CertificateForm.tsx

```diff
@@ -97,17 +112,68 @@ const CertificateForm: FC<{ initialValues: ICertificate }> = ({
   const handleClose = () => {
     setIsDialogOpen(false);
   };
+  const closeUserLookup = () => {
+    setIsUserLookup(false);
+  };
   const { translate } = useTranslate();
+  interface Column {
+    header: string;
+    accessor: keyof UserTable;
+  }
+  const columns: Column[] = useMemo(
+    () => [
+      { header: translate('Name'), accessor: 'name' },
+      { header: translate('Department'), accessor: 'department' },
+      { header: translate('Email'), accessor: 'email' },
+    ],
+    [],
+  );
 
+  const handleRemoveUser = (id: number) => {
+    if (
+      window.confirm(translate('Are you sure you want to remove this user?'))
+    ) {
+      setAssignedUsers((prevUsers: IUser[]) => {
+        try {
+          return prevUsers.filter((user) => user.id !== id);
+        } catch (error) {
+          console.error('Error removing user:', error);
+          return prevUsers;
+        }
+      });
+    }
+  };
+
+  const renderActions = (id: number) => (
+    <button
+      className="remove-user-button"
+      onClick={() => handleRemoveUser(id)}
+      aria-label={translate('Remove user')}
+    >
+      <img src={CloseIcon} alt={translate('Remove')} width={15} height={15} />
+    </button>
+  );
+  const tableData = useMemo(() => {
+    return assignedUsers.length > 0
+      ? assignedUsers
+      : formValues?.assignedUsers || [];
+  }, [assignedUsers, formValues?.assignedUsers]);
   return (
-    <section style={{ marginTop: '50px' }}>
-      <SearchCertificate
+    <section>
+      <SearchSupplier
         supplierName={supplierName || ''}
         setSupplierName={setSupplierName}
         handleDialogClose={handleClose}
         isDialogOpen={isDialogOpen}
         setIsDialogOpen={setIsDialogOpen}
       />
+      <UserLookup
+        assignedUsers={assignedUsers}
```

Comment:
This works, same as supplier search, but better approach would be getting rid of mid-state (assignedUsers, supplierName) and useEffects that observe changes to this state. Alternative approach would be that for both SearchSupplier and UserLookup, as handleClose you define function that accepts selected value as param. That way dialog can return you values and you can set them to formValues immediately.